### ISSUE TITLE
Unable to use with an ESP8266?

### ISSUE THREAD

**Issue by Darthmonkey** [hristo-atanasov/Tasmota-IRHVAC#145]
After flashing an ESP8266 with Tasmotizer, configuring the correct ports, adding an LED IR transmitter and receiver, and testing the remote talks to via the console (I don't know how to test the transmitter from the console), I followed the readme up to the point where it integrates into Home assistant.
My problem appears to be that I have a 'sensor' entity, rather than a 'climate' entity, which means I can't add a 'thermostat' to the Lovelace page. It may be that I've missed a step?
Should I have used an ESP32 instead, or am I missing something else?
Thanks in advance.

**Comment by cociweb** [comment]
> I followed the readme up to the point where it integrates into Home assistant.

Can you elaborate it please? It seems, you have missed something. (Whatever it is.... :))

**Comment by Darthmonkey** [comment]
### Just remember, you asked for elaboration... ;)
I'm attempting to use a Wemos D1 ESP8266 along with a KY-022, and up to 7 TSAL6400 LEDs with the Anodes attached to 220 ohm resistors to create a universal IR remote device which operates the same way as a Broadlink RM4 Mini, which can then be controlled exclusively via Home assistant.
I'm intending to either update this repos' readme or create a whole step by step instruction manual for people who may not intimately know all the various hurdles from hardware to software and be able to overcome them.
**My abridged steps so far are:**
1. Source a Wemos D1 Mini ESP8266, some KY-022 IR receivers, and a load of TSAL6400 LEDs with associated 220ohm resistors.
2. Connect the KY-022 to pin D3, and a single LED (for test purposes) to pin D4 via the resistor.
3. Flash the Wemos D1 with Tasmota4.1 (release-ir) 
4. Set up the LED sender on D3 GPIO0 to 'IRsend - 1'
5. Set up the KY-022 receiver on D4 GPIO2 to 'IRrecv'
6. Verify that the receiver can recognise my AC remote by pressing buttons and checking for a response on the devices' web page in the console.
7. Configure the Wemos to connect to HA via MQTT, naming it 'IRTest1' in the 'other' configuration.
8. Download the repo zip, and extract it to '...\config\Tasmota-IRHVAC-master' (I assume not '...\config\Tasmota-IRHVAC-master\Tasmota-IRHVAC-master' which is how it's packaged).
9. Using the console response from my remote in step 6, append the provided repo example, and edit the configuration.yaml on HA with the values where applicable, hashing out any that aren't, and noting any errors in the HA log files.
10. (This is where I'm getting unstuck). Edit the Home assistant lovelace page to assign a 'Thermostat card' to the entity (after a reboot) in this example the device is named 'IRtest1' which has a number of detected entity IDs, all starting 'sensor.x'
![image](https://github.com/hristo-atanasov/Tasmota-IRHVAC/assets/3304208/38e0ebe5-a83f-4db7-bb4a-57baca3e7782)
I'm _assuming_ that as shown in the example, the entity needs to start 'climate.x' rather than 'sensor.x' to be listed, (all the shown ones are from my Broadlink IR Sensors) or I missed something in the readme instructions which would mean I've deviated from the path somewhat.

**Comment by cociweb** [comment]
Great! Thank you!
> Using the console response from my remote in step 6, append the provided repo example, and edit the configuration.yaml on HA with the values where applicable, hashing out any that aren't, and noting any errors in the HA log files.

Can you please share the adjusted section from the configuration.yaml?

**Comment by Darthmonkey** [comment]
Even though I do remember doing the whole 'custom HACS installation from GitHub' thing previously, I had forgotten it was an option when I searched for it through HACS all three times. I've now installed via HACS, and I've been able to add it as a Thermostat card. Thanks for that.
I do have another two issues, which can be asked elsewhere if appropriate:
1. When I press a button on the thermostat and hold the device close to the AC it's not doing anything. I suspect that if I could send an ON/OFF via the Tasmota console, I might be able to troubleshoot, but I'm not sure how.
2. The console sometimes responds to inputs from the AC remote, probably every 10th button press, even though the light on the receiver flashes strongly when a button press is detected. Could that be a Wemos D1 issue?
[configuration.txt](https://github.com/user-attachments/files/15968618/configuration.txt)

configuration.yaml code attached if still relevant (it was pasted as code but formatting went wild).

**Comment by cociweb** [comment]
> probably every 10th button press

I'm not an expert of that, but there are low cost IR receivers/demodulators what are provide equivalent quality....
https://github.com/crankyoldgit/IRremoteESP8266/wiki/Frequently-Asked-Questions#help-im-getting-very-inconsistent-results-when-capturing-an-ir-message-using-a-vs1838b-ir-demodulator 
Is your IR demodulator matches with the one in the link?

> When I press a button on the thermostat and hold the device close to the AC it's not doing anything.

so you have IR modulator unit as well?

Additional reading in this HW topic is: https://tasmota.github.io/docs/IR-Remote/


When I've started from the beginning, first I've checked it with an example code of IRremoteESP8266 to observe that every feature/vendor/brand/model are correctly recognized because this library is included in tasmota.
ALL of the vendors/etc. are included only in the precompiled and purified Tasmota-IR variant. (or later, if you are interested only in your proto, then consider to build your own...)

Anyways, I also think that here we've solved your original problem, since this component is a bit more like a HA thermostat<=>mqtt with the payload of the correct IR messages.

**Comment by Darthmonkey** [comment]
Thanks very much for that, yes, I believe my journey continues, but not on this post.

**Comment by Darthmonkey** [comment]
Okay, it's still not _quite_ finished, but my preliminary tests have shown two things:
1. KISS is always the best method. I don't need 7 LEDs per IR sender, they're highly directional, but if one is pointed in the right direction, I've got a good 3 metres of range from one pin to one LED.
2. The ESP8266 Wemos D1 Mini is fine, and no resistors are required it is working (at least for the easy Panasonic AC)
3. Unless testing the remote signals, I don't even need to connect a receiver to the Wemos, which saves power, and provides more to the sender pin (I think).
To answer the question I posed, yes Tasmota-IRHVAC, does work with ESP8266.
